### ISSUE TITLE
Add the showCupertinoContextMenu function

### ISSUE THREAD

**Issue by laterdayi** [flutter/flutter#151807]
### Use case

Add the showCupertinoContextMenu function

Now that we have showMenu but it belongs to materialUI and not to Cupertino, we want to add a showCupertinoContextMenu function

### Proposal

Add the showCupertinoContextMenu function

Now that we have showMenu but it belongs to materialUI and not to Cupertino, we want to add a showCupertinoContextMenu function

**Comment by huycozy** [comment]
Hi @laterdayi 
You are duplicating `Use case` and `Proposal` section, can you please be more specific on the `Use case`? Doesn't [CupertinoContextMenu](https://api.flutter.dev/flutter/cupertino/CupertinoContextMenu-class.html) fit your case?

**Comment by laterdayi** [comment]
> Hi @laterdayi You are duplicating `Use case` and `Proposal` section, can you please be more specific on the `Use case`? Doesn't [CupertinoContextMenu](https://api.flutter.dev/flutter/cupertino/CupertinoContextMenu-class.html) fit your case?

I have a scenario where I want the CupertinoContextMenu to pop up when onLongPress of the dataTable DataRow, so I need a showCupertinoContextMenu method, just like showMenu

**Comment by huycozy** [comment]
Looks like the context is the same at https://github.com/flutter/flutter/issues/151745. In such a case, we can't use CupertinoContextMenu as a DataRow type. Labeling this as a feature request.

**Comment by laterdayi** [comment]
Yes, a new method needs to be added to implement it, looking forward to adding

**Comment by MaherSafadii** [comment]
check out #143712

**Comment by laterdayi** [comment]
> check out #143712

This seems to be a component, not a function call display

**Comment by justinmc** [comment]
@laterdayi You don't want the usual CupertinoContextMenu bounce animation of the longpress target, you just want to immediately open it? Is there still a target widget that animates into the new route, or the new route just appears without animation?

Here's the code where CupertinoContextMenu is opened: https://github.com/flutter/flutter/blob/978657becd854b0b5ced80d372bdfc9b82da19b0/packages/flutter/lib/src/cupertino/context_menu.dart#L464

It's not currently written in a way that would be easy to abstract into a global function. I guess you can't use CupertinoContextMenu with the DataRow for some reason?

**Comment by laterdayi** [comment]
> @laterdayi You don't want the usual CupertinoContextMenu bounce animation of the longpress target, you just want to immediately open it? Is there still a target widget that animates into the new route, or the new route just appears without animation?
> 
> Here's the code where CupertinoContextMenu is opened:
> 
> https://github.com/flutter/flutter/blob/978657becd854b0b5ced80d372bdfc9b82da19b0/packages/flutter/lib/src/cupertino/context_menu.dart#L464
> 
> It's not currently written in a way that would be easy to abstract into a global function. I guess you can't use CupertinoContextMenu with the DataRow for some reason?
> 
> I'm thinking this proposal might not be practical, but we need some more information to understand what it would do.

       In many scenarios, we may not need a component to pop up, but need to open the context menu through a function method, which should be a very common scenario
